scanpy/tools/_paga.py

```diff
@@ -534,24 +533,24 @@ def paga_compare_paths(
             n_steps += 1
             continue
         if len(path1) >= len(path2):
-            path_mapped = [asso_groups1[l] for l in path1]
+            path_mapped = [asso_groups1[link] for link in path1]
             path_compare = path2
             path_compare_id = 2
             path_compare_orig_names = [
-                [orig_names2[int(s)] for s in l] for l in path_compare
+                [orig_names2[int(s)] for s in link] for link in path_compare
             ]
             path_mapped_orig_names = [
-                [orig_names2[int(s)] for s in l] for l in path_mapped
+                [orig_names2[int(s)] for s in link] for link in path_mapped
             ]
         else:
-            path_mapped = [asso_groups2[l] for l in path2]
+            path_mapped = [asso_groups2[link] for link in path2]
             path_compare = path1
             path_compare_id = 1
             path_compare_orig_names = [
-                [orig_names1[int(s)] for s in l] for l in path_compare
+                [orig_names1[int(s)] for s in link] for link in path_compare
             ]
             path_mapped_orig_names = [
-                [orig_names1[int(s)] for s in l] for l in path_mapped
+                [orig_names1[int(s)] for s in link] for link in path_mapped
```

Comment:
They are redefinitions and should not stay as `l`. I'd be fine with either

1. renaming to _
2. renaming to k/j or something
3. Opening a follow up issue for this.

Reply:
I'm not seeing how these are redefinitions, as comprehensions get their own scope. Am I missing something? If anything, wouldn't `s` be the redefinition?

> renaming to _

This would be wrong, since the value is used, right?

> renaming to k/j or something

If there really is a problem with `l`, then I'm fairly confident this variable refers to `leaf`s. My preference is still not to touch it.

Reply:
I honestly don't see it either, but Pycharm and flake8 agree that these are redefinitions. Maybe there's a global imported `l` somewhere. I went with leafs now.